server/server.js

```diff
@@ -20,14 +21,38 @@ var Donation    = require('./models/donation').Donation;
 // CONFIGURE APP
 // =============================================================================
 var app         = express();
-var api         = require ('./routes/api');
-var webhooks    = require('./routes/webhooks');
+
+app.use((req, res, next) => {
+
+    req.rawBody = '';
+    req.on('data', function(chunk) {
+        req.rawBody += chunk;
+    });
+
+    req.on('end', function() {
+        // Validate shopify webhook token. If the hashes don't match, reject the request
+        if (/.*\/shopify\/.*/.test(req.url)) {
```

Comment:
What is .test ?

Reply:
I mean test()

Reply:
Dumb name for what it is. All it does is check the input string against the regex and returns true or false. “Test” makes it seem like it’s part of some unit test library or something. But it’s a vanilla JavaScript function.